### ISSUE TITLE
Cannot create github_branch_protection_v3 after update to v5.3.0

### ISSUE THREAD

**Issue by jtsaito** [integrations/terraform-provider-github#1307]
### Terraform Version
**v5.3.0**

### Affected Resource(s)

- github_branch_protection_v3
- github_branch_default
- github_branch_default

The issue occures when creating a new `github_branch_protection_v3` resource with attributes referenced from `github_branch_default` and `github_branch_protection_v3` resources.

### Debug Output

```
github_branch_protection_v3.<MY_REPO_NAME_REDACTED>: Creating...
╷
│ Error: PUT https://api.github.com/repos/<MY_ORG_REDACTED>/<MY_REPO_NAME_REDACTED>/branches/main/protection: 422 Invalid request.
│
│ No subschema in "anyOf" matched.
│ For 'properties/checks', nil is not an array.
│ Not all subschemas of "allOf" matched.
│ For 'anyOf/1', {"strict"=>true, "contexts"=>["build"], "checks"=>nil} is not a null. []
```

### Expected Behavior
The branch protection v3 resource should be created.

### Actual Behavior
Terraform errors out with message above.

#### Step 1

Create a repo with terraform apply.

```
resource "github_repository" "test-repo" {
  name        = "test-repo"
  description = "This repo is a test"

  has_downloads = true
  has_issues    = false
  has_projects  = false
  has_wiki      = false

  visibility = "private"

  auto_init = true

  vulnerability_alerts = false
}

resource "github_branch_default" "test-repo" {
  repository = github_repository.test-repo.name
  branch     = "main"
}
```

#### Step 2

Terraform apply the following change:

```
resource "github_branch_protection_v3" "test-repo" {
  repository     = github_repository.test-repo.name
  branch         = github_branch_default.test-repo.branch
  enforce_admins = true

  required_status_checks {
    strict   = true
    contexts = ["build"]
  }

  required_pull_request_reviews {
    dismiss_stale_reviews      = true
    require_code_owner_reviews = true
  }
}
```

### Important Factoids
This works up to GitHub provider version v5.2.0 without errors and errors in v5.3.0.

This seems to be linked to the go client update in the [release diff](https://github.com/integrations/terraform-provider-github/compare/v5.2.0...v5.3.0)  where the API Client is updated as follows.

https://github.com/integrations/terraform-provider-github/blob/6773d80600f761bf50b695f475c2d457d1151ee5/vendor/github.com/google/go-github/v47/github/repos.go#L908

From 

```
Checks []*RequiredStatusCheck `json:"checks,omitempty"
```

To

```
Checks []*RequiredStatusCheck `json:"checks"`
```

### References
- https://github.com/integrations/terraform-provider-github/releases

**Comment by iniinikoski** [comment]
Can confirm this. Thanks for reporting @jtsaito !

**Comment by jtsaito** [comment]
@iniinikoski Thanks for looking into it and confirming the bug.

**Comment by rpdelaney** [comment]
Here's the error message we're getting:

```
╷
│ Error: PUT https://api.github.com/repos/trussworks/terraform-aws-route53-query-logs/branches/main/protection: 422 Invalid request.
│
│ No subschema in "anyOf" matched.
│ For 'properties/checks', nil is not an array.
│ Not all subschemas of "allOf" matched.
│ For 'anyOf/1', {"strict"=>true, "contexts"=>["validate"], "checks"=>nil} is not a null. []
│
│   with github_branch_protection_v3.terraform_aws_route53_query_logs_main,
│   on trussworks-public-repos.tf line 4700, in resource "github_branch_protection_v3" "terraform_aws_route53_query_logs_main":
│ 4700: resource "github_branch_protection_v3" "terraform_aws_route53_query_logs_main" {
│
╵
╷
│ Error: PUT https://api.github.com/repos/trussworks/terraform-aws-waf/branches/main/protection: 422 Invalid request.
│
│ No subschema in "anyOf" matched.
│ For 'properties/checks', nil is not an array.
│ Not all subschemas of "allOf" matched.
│ For 'anyOf/1', {"strict"=>true, "contexts"=>["validate"], "checks"=>nil} is not a null. []
│
│   with github_branch_protection_v3.terraform_aws_waf_main,
│   on trussworks-public-repos.tf line 4870, in resource "github_branch_protection_v3" "terraform_aws_waf_main":
│ 4870: resource "github_branch_protection_v3" "terraform_aws_waf_main" {
│
╵
```

**Comment by soerenmartius** [comment]
same here

**Comment by TheQueenIsDead** [comment]
@kfcampbell I've opened a draft PR with a potential solution here:
https://github.com/integrations/terraform-provider-github/pull/1409

It simply adds in an empty for the missing `Checks` atrtibute, which should resolve the issue as proposed by the issue reporter in https://github.com/google/go-github/issues/2467

If this seems like a viable path forwards, I'll look at building the module and testing it when I get the chance :-) (Untested currently hence the draft)

**Comment by kfcampbell** [comment]
I responded on the PR, which looks good! I can't wait to resolve this issue.

**Comment by jtsaito** [comment]
@TheQueenIsDead  and everyone else who contributed. The issue was fixed by release `5.16.0` which included https://github.com/integrations/terraform-provider-github/pull/1415.

**Comment by ekhaydarov** [comment]
im not so sure that this works? I am on 5.16.0 and hit the nil issue, adding checks just gave me this
```
Error: Unsupported argument 
...
An argument named "checks" is not expected here.
```
